netlink/conntracker.go

```diff
@@ -0,0 +1,305 @@
+// +build linux
+
+package netlink
+
+import (
+	"bytes"
+	"context"
+	golog "log"
+	"net"
+	"os"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	ct "github.com/florianl/go-conntrack"
+)
+
+// Conntracker is a wrapper around go-conntracker that keeps a record of all connections in user space
+type Conntracker interface {
+	GetTranslationForConn(ip string, port uint16) *IPTranslation
+	ClearShortLived()
+	Close()
+}
+
+type connKey struct {
+	ip   string
+	port uint16
+}
+
+type realConntracker struct {
+	sync.Mutex
+	nfct  *ct.Nfct
+	state map[connKey]*IPTranslation
+
+	// a short term buffer of connections to IPTranslations. Since we cannot make sure that tracer.go
+	// will try to read the translation for an IP before the delete callback happens, we will
+	// safe a fixed number of connections
+	shortLivedBuffer map[connKey]*IPTranslation
+
+	// the maximum size of the short lived buffer
+	maxShortLivedBuffer int
+
+	statsTicker   *time.Ticker
+	compactTicker *time.Ticker
+	stats         struct {
+		gets               int64
+		getTimeTotal       int64
+		registers          int64
+		registersTotalTime int64
+	}
+}
+
+func NewConntracker() (Conntracker, error) {
+	nfct, err := ct.Open(&ct.Config{ReadTimeout: 10 * time.Millisecond, Logger: golog.New(os.Stdout, "go-conntrack", 0)})
```

Comment:
What effect is this logger going to have on our current logging library? Can we redirect them to it somehow?

Reply:
I had enabled this early on for testing. In practice it wasn't helpful, so I've removed it for now.